packages/pharos/src/components/toast/pharos-toaster.ts

```diff
@@ -38,29 +39,46 @@ export class PharosToaster extends PharosElement {
   override connectedCallback(): void {
     super.connectedCallback && super.connectedCallback();
     document.addEventListener('pharos-toast-open', this._openToast as EventListener);
+    document.addEventListener('pharos-toast-update', this._updateToast as EventListener);
     document.addEventListener('pharos-toast-close', this._closeToast as EventListener);
   }
 
   override disconnectedCallback(): void {
     document.removeEventListener('pharos-toast-open', this._openToast as EventListener);
+    document.removeEventListener('pharos-toast-update', this._updateToast as EventListener);
     document.removeEventListener('pharos-toast-close', this._closeToast as EventListener);
     super.disconnectedCallback && super.disconnectedCallback();
   }
 
   private async _openToast(event: Event): Promise<void> {
     const toastTag = this.localName.split('pharos-toaster')[0] + 'pharos-toast';
     const toast = document.createElement(toastTag) as PharosToast;
-    const { content, status } = (<CustomEvent>event).detail;
+    const { content, status, id, indefinite } = (<CustomEvent>event).detail;
 
     toast.innerHTML = content;
     toast.status = status || DEFAULT_STATUS;
+    toast.id = id || DEFAULT_ID;
+    toast.indefinite = indefinite || DEFAULT_INDEFINITE;
     this.insertBefore(toast, this.childNodes[0] || null);
     await this.updateComplete;
     toast.focus();
   }
 
+  private _updateToast(event: CustomEvent): void {
+    const { content, status, id } = (<CustomEvent>event).detail;
+    const toast = document.getElementById(id || DEFAULT_ID);
+    if (toast) {
+      toast.innerHTML = content;
+      toast.status = status;
+    }
+  }
+
   private _closeToast(event: CustomEvent): void {
-    this.removeChild(event.detail);
+    const { id } = (<CustomEvent>event).detail;
+    const toast = document.getElementById(id || DEFAULT_ID);
```

Comment:
The concept of "the toast ID if provided, otherwise the default toast ID" appears a few times in the code—could be worth extracting for consistency